### ISSUE TITLE
this.getUrl() in Custom Commands doesn't return value, but title: Promise { <pending> }

### ISSUE THREAD

**Issue by rdpanek** [webdriverio/webdriverio#3209]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.1.0
 - **Mode:** [Testrunner]
 - **If WDIO Testrunner, running sync/async:** sync: true
 - **Node.js version:** 10.9.0
 - **NPM version:** 6.2.0
 - **Browser name and version:** Chrome 71
 - **Platform name and version:** OSX Mojave 10.14.1 
 - **Additional wdio packages used (if applicable):
```
"devDependencies": {
    "@babel/cli": "^7.2.3",
    "@babel/core": "^7.2.2",
    "@babel/preset-env": "^7.2.3",
    "@babel/register": "^7.0.0",
    "@wdio/cli": "^5.0.3",
    "@wdio/local-runner": "^5.1.0",
    "@wdio/mocha-framework": "^5.1.0",
    "@wdio/sync": "^5.1.0",
    "aws-sdk": "2.0.9",
    "chai": "4.1.2",
    "node-uuid": "1.4.8",
    "webdriverio": "^5.1.0"
  }
```

**Config of WebdriverIO**
wdio.conf.js
```
exports.config = {
    sync: true,
...

before: function (capabilities, specs) {
       browser.addCommand("getUrlAndTitle", function() {
          return {
              url: this.getUrl(),
              title: this.getTitle()
          };
      });
     },
...
```
**Describe the bug**
Example in this https://webdriver.io/docs/customcommands.html page doesn't work well
On stdout is
```
[0-0] { url: Promise { <pending> }, title: Promise { <pending> } }
```

**To Reproduce**

My wdio.conf.js contains in before hook customCommand `getUrlAndTitle` and in test file contains
```
...
it('someTestStep', function(){
    console.log(browser.getUrlAndTitle())
  });
...
```

**Expected behavior**
value from custom command should be title and url. In WDIO v4 it works well

**Log**
Selenium
```
23:26:54.512 DEBUG [WebDriverServlet.handle] - Found handler: org.openqa.selenium.remote.server.commandhandler.BeginSession@71b9d3dc
23:26:54.515 DEBUG [WebDriverServlet.lambda$handle$1] - /session: Executing POST on /session (handler: BeginSession)
23:26:54.602 INFO [ActiveSessionFactory.apply] - Capabilities are: Capabilities {browserName: chrome}
23:26:54.603 INFO [ActiveSessionFactory.lambda$apply$11] - Matched factory org.openqa.selenium.remote.server.ServicedSession$Factory (provider: org.openqa.selenium.chrome.ChromeDriverService)
23:26:54.653 DEBUG [UrlChecker.waitUntilAvailable] - Waiting for [http://localhost:8331/status]
23:26:54.654 DEBUG [UrlChecker.lambda$waitUntilAvailable$1] - Polling http://localhost:8331/status
23:26:54.673 DEBUG [UrlChecker.lambda$waitUntilAvailable$1] - Polling http://localhost:8331/status
23:26:54.698 DEBUG [UrlChecker.lambda$waitUntilAvailable$1] - Polling http://localhost:8331/status
Starting ChromeDriver 2.39.562713 (dd642283e958a93ebf6891600db055f1f1b4f3b2) on port 8331
Only local connections are allowed.
23:26:54.743 DEBUG [UrlChecker.lambda$waitUntilAvailable$1] - Polling http://localhost:8331/status
23:26:54.829 DEBUG [UrlChecker.lambda$waitUntilAvailable$1] - Polling http://localhost:8331/status
23:26:54.831 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@d45237c5 pairs: {GET /status HTTP/1.1: null}{User-Agent: Java/10.0.1}{Host: localhost:8331}{Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2}{Connection: keep-alive}
23:26:54.854 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@61dc8d603 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 126}{Content-Type: application/json; charset=utf-8}
23:26:56.326 INFO [ProtocolHandshake.createSession] - Detected dialect: OSS
23:26:56.346 DEBUG [RemoteWebDriver.log] - Executing: newSession [null, newSession {desiredCapabilities=Capabilities {acceptInsecureCerts: false, acceptSslCerts: false, applicationCacheEnabled: false, browserConnectionEnabled: false, browserName: chrome, chrome: {chromedriverVersion: 2.39.562713 (dd642283e958a9..., userDataDir: /var/folders/9d/q75r4gy51p5...}, cssSelectorsEnabled: true, databaseEnabled: false, handlesAlerts: true, hasTouchScreen: false, javascriptEnabled: true, locationContextEnabled: true, mobileEmulationEnabled: false, nativeEvents: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: MAC, rotatable: false, setWindowRect: true, takesHeapSnapshot: true, takesScreenshot: true, unexpectedAlertBehaviour: , unhandledPromptBehavior: , version: 71.0.3578.98, webStorageEnabled: true}}]
23:26:56.347 DEBUG [RemoteWebDriver.log] - Executed: [null, newSession {desiredCapabilities=Capabilities {acceptInsecureCerts: false, acceptSslCerts: false, applicationCacheEnabled: false, browserConnectionEnabled: false, browserName: chrome, chrome: {chromedriverVersion: 2.39.562713 (dd642283e958a9..., userDataDir: /var/folders/9d/q75r4gy51p5...}, cssSelectorsEnabled: true, databaseEnabled: false, handlesAlerts: true, hasTouchScreen: false, javascriptEnabled: true, locationContextEnabled: true, mobileEmulationEnabled: false, nativeEvents: true, networkConnectionEnabled: false, pageLoadStrategy: normal, platform: MAC, rotatable: false, setWindowRect: true, takesHeapSnapshot: true, takesScreenshot: true, unexpectedAlertBehaviour: , unhandledPromptBehavior: , version: 71.0.3578.98, webStorageEnabled: true}}]
23:26:56.672 INFO [RemoteSession$Factory.lambda$performHandshake$0] - Started new session 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:26:57.299 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:26:57.299 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing POST on /session/477652919367499589bde10e83027d2a/url (handler: ServicedSession)
23:26:57.302 DEBUG [Passthrough.handle] - To upstream: {"url":"https://www.XZY.cz/produkty/bezny-ucet/"}
23:26:57.305 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@54d489929 pairs: {POST /session/477652919367499589bde10e83027d2a/url HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Content-Type: application/json; charset=utf-8}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}{Content-Length: 53}
23:27:00.765 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@3772d5a94 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 72}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:00.770 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":null}
23:27:00.777 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:00.777 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing POST on /session/477652919367499589bde10e83027d2a/element (handler: ServicedSession)
23:27:00.779 DEBUG [Passthrough.handle] - To upstream: {"using":"xpath","value":"//title[contains(text(),'Běžný účet zdarma s úročením')]"}
23:27:00.779 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@153ec5e59 pairs: {POST /session/477652919367499589bde10e83027d2a/element HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Content-Type: application/json; charset=utf-8}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}{Content-Length: 92}
23:27:00.800 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@4dc4b3094 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 102}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:00.800 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":{"ELEMENT":"0.1984700775074717-1"}}
23:27:00.860 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:00.860 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing POST on /session/477652919367499589bde10e83027d2a/elements (handler: ServicedSession)
23:27:00.862 DEBUG [Passthrough.handle] - To upstream: {"using":"xpath","value":"//title[contains(text(),'Běžný účet zdarma s úročením')]"}
23:27:00.862 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@19affc7e9 pairs: {POST /session/477652919367499589bde10e83027d2a/elements HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Content-Type: application/json; charset=utf-8}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}{Content-Length: 92}
23:27:01.095 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@7455a23a4 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 104}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:01.096 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":[{"ELEMENT":"0.1984700775074717-1"}]}
23:27:01.105 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:01.106 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing POST on /session/477652919367499589bde10e83027d2a/element (handler: ServicedSession)
23:27:01.107 DEBUG [Passthrough.handle] - To upstream: {"using":"xpath","value":"//a/span[contains(text(),\"Založit účet\")]"}
23:27:01.108 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@17a740b89 pairs: {POST /session/477652919367499589bde10e83027d2a/element HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Content-Type: application/json; charset=utf-8}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}{Content-Length: 74}
23:27:01.128 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@7f13e5be4 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 102}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:01.128 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":{"ELEMENT":"0.1984700775074717-2"}}
23:27:01.134 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:01.134 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing POST on /session/477652919367499589bde10e83027d2a/elements (handler: ServicedSession)
23:27:01.136 DEBUG [Passthrough.handle] - To upstream: {"using":"xpath","value":"//a/span[contains(text(),\"Založit účet\")]"}
23:27:01.136 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@19b9ba2e9 pairs: {POST /session/477652919367499589bde10e83027d2a/elements HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Content-Type: application/json; charset=utf-8}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}{Content-Length: 74}
23:27:01.155 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@40a48f514 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 139}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:01.156 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":[{"ELEMENT":"0.1984700775074717-2"},{"ELEMENT":"0.1984700775074717-3"}]}
23:27:01.167 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:01.168 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing GET on /session/477652919367499589bde10e83027d2a/element/0.1984700775074717-2/displayed (handler: ServicedSession)
23:27:01.170 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@6d2ba57d7 pairs: {GET /session/477652919367499589bde10e83027d2a/element/0.1984700775074717-2/displayed HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}
23:27:01.200 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@d7583734 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 72}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:01.200 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":true}
23:27:01.222 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:01.223 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing GET on /session/477652919367499589bde10e83027d2a/url (handler: ServicedSession)
23:27:01.225 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@59c920397 pairs: {GET /session/477652919367499589bde10e83027d2a/url HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}
23:27:01.231 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:01.232 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing GET on /session/477652919367499589bde10e83027d2a/title (handler: ServicedSession)
23:27:01.233 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@47f318127 pairs: {GET /session/477652919367499589bde10e83027d2a/title HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}
23:27:01.238 DEBUG [WebDriverServlet.handle] - Found handler: 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:01.239 DEBUG [WebDriverServlet.lambda$handle$1] - Handler thread for session 477652919367499589bde10e83027d2a (chrome): Executing DELETE on /session/477652919367499589bde10e83027d2a (handler: ServicedSession)
23:27:01.240 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@7bbcda847 pairs: {DELETE /session/477652919367499589bde10e83027d2a HTTP/1.1: null}{User-Agent: webdriver/5.1.0}{Accept: application/json}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{Host: localhost:8331}
23:27:01.258 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@3d73e2564 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 113}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:01.259 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":"https://www.XZY.cz/produkty/bezny-ucet/"}
23:27:01.294 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@4934ee9c4 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 123}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:01.296 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":"Běžný účet zdarma s úročením 1 % • XYZ"}
23:27:01.346 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@757b3b834 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 72}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:01.347 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"477652919367499589bde10e83027d2a","status":0,"value":null}
23:27:01.355 INFO [ActiveSessions$1.onStop] - Removing session 477652919367499589bde10e83027d2a (org.openqa.selenium.chrome.ChromeDriverService)
23:27:01.366 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@126d5f227 pairs: {DELETE /session/477652919367499589bde10e83027d2a HTTP/1.1: null}{Connection: close}{Cache-Control: no-cache}{Pragma: no-cache}{User-Agent: Java/10.0.1}{Host: localhost:8331}{Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2}
23:27:01.388 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@663510a44 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 40}{Content-Type: application/json; charset=utf-8}{Connection: close}
23:27:01.389 DEBUG [Passthrough.handle] - To downstream: {"sessionId":"","status":0,"value":null}
23:27:01.390 DEBUG [UrlChecker.waitUntilUnavailable] - Waiting for http://localhost:8331/shutdown
23:27:01.391 DEBUG [UrlChecker.lambda$waitUntilUnavailable$2] - Polling http://localhost:8331/shutdown
23:27:01.391 DEBUG [HttpURLConnection.writeRequests] - sun.net.www.MessageHeader@457ec06f5 pairs: {GET /shutdown HTTP/1.1: null}{User-Agent: Java/10.0.1}{Host: localhost:8331}{Accept: text/html, image/gif, image/jpeg, *; q=.2, */*; q=.2}{Connection: keep-alive}
23:27:01.392 DEBUG [HttpURLConnection.getInputStream0] - sun.net.www.MessageHeader@72605b663 pairs: {null: HTTP/1.1 200 OK}{Content-Length: 40}{Content-Type: application/json; charset=utf-8}
23:27:01.404 DEBUG [UrlChecker.lambda$waitUntilUnavailable$2] - Polling http://localhost:8331/shutdown
```

WDIO
```
...
[0-0] 2018-12-29T22:07:53.169Z INFO webdriver: RESULT true
[0-0] 2018-12-29T22:07:53.169Z INFO webdriver: COMMAND getUrlAndTitle()
[0-0] 2018-12-29T22:07:53.170Z INFO webdriver: COMMAND getUrl()
[0-0] 2018-12-29T22:07:53.170Z INFO webdriver: [GET] http://127.0.0.1:4444/wd/hub/session/68f243dbb6774ff5d7a2d481501b0710/url
[0-0] 2018-12-29T22:07:53.171Z INFO webdriver: COMMAND getTitle()
[0-0] 2018-12-29T22:07:53.171Z INFO webdriver: [GET] http://127.0.0.1:4444/wd/hub/session/68f243dbb6774ff5d7a2d481501b0710/title
[0-0] { url: Promise { <pending> }, title: Promise { <pending> } }
[0-0] 2018-12-29T22:07:53.172Z INFO webdriver: RESULT { url: Promise { <pending> }, title: Promise { <pending> } }
[0-0] 2018-12-29T22:07:53.177Z INFO webdriver: COMMAND deleteSession()
2018-12-29T22:07:53.177Z INFO webdriver: [DELETE] http://127.0.0.1:4444/wd/hub/session/68f243dbb6774ff5d7a2d481501b0710
...
```

**Additional context**

> Please use the correct markdown

**Comment by christian-bromann** [comment]
@rdpanek this is a bug, if you use sync you shouldn't need to use async await in custom commands.

**Comment by jdavis61** [comment]
Hi @rdpanek I believe the issue in this ticket was fixed in version 5.2.8. After upgrading to this version, any custom command with async/await failed when sync = true is configured in wdio.conf.js.

For example when sync=true, this command completely failed.
```
browser.addCommand('element', async selector => $(selector));
```

This command did not throw an error.
```
browser.addCommand('element', selector => $(selector));
```

FYI

**Comment by christian-bromann** [comment]
This has been fixed with the latest release.
